rust/worker/src/utils/convert.rs

```diff
@@ -0,0 +1,165 @@
+use chroma_types::{
+    chroma_proto::{self, GetResult, KnnBatchResult, KnnResult},
+    ScalarEncoding, Where,
+};
+use tonic::Status;
+
+use crate::execution::operators::{
+    filter::FilterOperator,
+    knn::KnnOperator,
+    knn_projection::{KnnProjectionOperator, KnnProjectionOutput, KnnProjectionRecord},
+    limit::LimitOperator,
+    projection::{ProjectionOperator, ProjectionOutput, ProjectionRecord},
+};
+
+impl TryFrom<chroma_proto::FilterOperator> for FilterOperator {
+    type Error = Status;
+
+    fn try_from(value: chroma_proto::FilterOperator) -> Result<Self, Status> {
```

Comment:
this really ought to be chroma errors - its the job of the server to know about `Status` and to perform that conversion

Reply:
Updated to `ConversionError` in `chroma_type` crate